### PR TITLE
Move provider's folder of currency to currency/provider

### PR DESCRIPTION
**Pull request by younies**
Under dimension, I would like to have a folder for each component, `currency`, `units` and `percent`

Therefore, I moved the `provider` folder under `currency`. In the next PR: I will move percent's files into dimension/percent forlder.

### CODE REVIEW

**Review by robertbastian — CHANGES_REQUESTED** [review]
We usually have a provider module at the top level, see https://github.com/unicode-org/icu4x/tree/main/components/locid_transform/src. Generated constructors for example expect this.

File: provider/datagen/tests/data/baked/macros/percent_essentials_v1.rs.data [hunk 1]
```diff
@@ -9,15 +9,15 @@ macro_rules! __impl_percent_essentials_v1 {
...
-        impl icu_provider::DataProvider<icu::experimental::dimension::provider::percent::PercentEssentialsV1Marker> for $provider {
-            fn load(&self, req: icu_provider::DataRequest) -> Result<icu_provider::DataResponse<icu::experimental::dimension::provider::percent::PercentEssentialsV1Marker>, icu_provider::DataError> {
-                static UND: <icu::experimental::dimension::provider::percent::PercentEssentialsV1Marker as icu_provider::DataMarker>::Yokeable = icu::experimental::dimension::provider::percent::PercentEssentialsV1 { number_index: 0u8, percent_sign_affixes: icu::experimental::dimension::provider::percent::PercentAffixesV1 { prefix: alloc::borrow::Cow::Borrowed(""), suffix: alloc::borrow::Cow::Borrowed("") }, percent_sign_symbol: alloc::borrow::Cow::Borrowed("%"), percent_symbol_index: 5u8, standard: alloc::borrow::Cow::Borrowed("#,##0%") };
-                static VALUES: [&<icu::experimental::dimension::provider::percent::PercentEssentialsV1Marker as icu_provider::DataMarker>::Yokeable; 1usize] = [&UND];
+        impl icu_provider::DataProvider<icu::experimental::dimension::currency::provider::percent::PercentEssentialsV1Marker> for $provider {
```
Comment (younies) [inline provider/datagen/tests/data/baked/macros/percent_essentials_v1.rs.data:12]: @robertbastian : In this file, I have to make the changes manually, even though, these changes should came from databake ? do you have any thoughts about this ?

Comment (robertbastian) [reply provider/datagen/tests/data/baked/macros/percent_essentials_v1.rs.data:12]: this is generated with `cargo make testdata`

**Comment by sffc** [comment]
Defer to @robertbastian, and noting that in other components, we add more modules under `provider`

**Comment by robertbastian** [comment]
I think a single provider module has advantages, like keeping all unstable code in one place, and being easy to locate, so let's follow this convention for dimension as well.

**Comment by younies** [comment]
> I think a single provider module has advantages, like keeping all unstable code in one place, and being easy to locate, so let's follow this convention for dimension as well.

OKay, no problem

**Pull request closed without merging**